### PR TITLE
Update fake stackdriver to response audit log entry separately.

### PR DESCRIPTION
**Pull request by bianpengyuan**
cc @xulingqing

### CODE REVIEW

File: test/envoye2e/stackdriver_plugin/fake_stackdriver.go [hunk 1]
```diff
@@ -484,6 +502,7 @@ func RunFakeStackdriver(port uint16) error {
 	}
 	http.HandleFunc("/timeseries", fsdms.GetTimeSeries)
 	http.HandleFunc("/logentries", fsdls.GetLogEntries)
+	http.HandleFunc("/auditlogentries", fsdls.GetAuditLogEntries)
```
Comment (gargnupur) [inline test/envoye2e/stackdriver_plugin/fake_stackdriver.go:505]: why not add the log name in log entry? These are just server entries.. I think we would want to add client side logs too in future and then we would have to add more rest url's for that?

Comment (bianpengyuan) [reply test/envoye2e/stackdriver_plugin/fake_stackdriver.go]: Done.

File: test/envoye2e/stackdriver_plugin/fake_stackdriver.go [hunk 2]
```diff
@@ -143,11 +143,12 @@ func (s *LoggingServer) WriteLogEntries(ctx context.Context, req *logging.WriteL
 	s.mux.Lock()
 	defer s.mux.Unlock()
 	for _, entry := range req.Entries {
-		// Add the general labels to every log entry in list logentries response.
+		// Add the general labels and log name to every log entry in list logentries response.
 		tmpEntry := proto.Clone(entry).(*logging.LogEntry)
 		for k, v := range req.Labels {
 			tmpEntry.Labels[k] = v
 		}
+		tmpEntry.Labels["log_name"] = req.LogName
```
Comment (gargnupur) [inline test/envoye2e/stackdriver_plugin/fake_stackdriver.go:151]: We could also use `LogName` field in LogEntry(https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry) ?

Comment (bianpengyuan) [reply test/envoye2e/stackdriver_plugin/fake_stackdriver.go]: Thanks done!

**Review by gargnupur — APPROVED** [review]

**Pull request merged**